src/pages/index.js

```diff
@@ -30,55 +36,202 @@ const addCardPopup = new PopupWithForm(
   "#profile__add-card-modal",
   handleAddCardSubmit
 );
+const editAvatarPopup = new PopupWithForm(
+  "#profile__avatar-modal",
+  handleAvatarSubmit
+);
 const previewImagePopup = new PopupWithImage("#card_modal");
+const trashConfirmPopup = new PopupWithConfirm("#trashcan-modal");
+
+//Modal Event Listeners
+profileEditPopup.setEventListeners();
+addCardPopup.setEventListeners();
+editAvatarPopup.setEventListeners();
+trashConfirmPopup.setEventListeners();
 
 // Section
-const cardSection = new Section(
-  { items: initialCards, renderer },
-  ".cards__list"
-);
-cardSection.renderItems();
+const cardSection = new Section({ items: [], renderer }, ".cards__list");
+
+// Validators
+const profileEditForm = document.querySelector("#profileEditForm");
+const addCardForm = document.querySelector("#addcard__form");
+const avatarForm = document.querySelector("#changeprofile_form");
 
-// Form validators
 const profileEditFormValidator = new FormValidator(config, profileEditForm);
-const addCardFormValidator = new FormValidator(config, addCardForm);
 profileEditFormValidator.enableValidation();
+const addCardFormValidator = new FormValidator(config, addCardForm);
 addCardFormValidator.enableValidation();
+const avatarFormValidator = new FormValidator(config, avatarForm);
+avatarFormValidator.enableValidation();
 
-// Event Listeners
-profileEditBtn.addEventListener("click", () => {
-  const userData = userInfo.getUserInfo();
-  profileTitleInput.value = userData.name;
-  profileDescriptionInput.value = userData.about;
-  profileEditPopup.open();
-});
-
-addCardButton.addEventListener("click", () => addCardPopup.open());
+// Constants
+const profileEditBtn = document.querySelector("#profile__edit-button");
+const addCardButton = document.querySelector(".profile__add-button");
+const profileTitleInput = document.querySelector("#profile__name-input");
+const profileDescriptionInput = document.querySelector(
+  "#profile__subheading-input"
+);
+const editProfileImage = document.querySelector(".profile__avatar-edit");
 
-// Event handlers
+// Functions
 function handleProfileEditSubmit(profileData) {
-  const { modal__input_type_name: name, modal__input_type_description: about } =
-    profileData;
-  userInfo.setUserInfo(name, about);
-  profileEditPopup.close();
+  const name = profileData.modal__input_type_name;
+  const about = profileData.modal__input_type_description;
+
+  profileEditPopup.renderLoading(true);
+
+  return api
+    .patchProfileInfo(name, about)
+    .then(() => {
+      userInfo.setUserInfo(name, about);
+      profileEditPopup.close();
+      profileEditForm.reset();
+      profileEditFormValidator._disableSubmitButton();
```

Comment:
`_disableSubmitButton`

Please, delete the **low dash** (`underscore`) from methods/fields if you want to use them **outside** a class because the `low dash` shows a **private** field which is used only inside the class.

**Please, correct it everywhere in the code**